packages/react-components/src/graph-visualizer/cytoGraph.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import type { LayoutOptions } from 'cytoscape';
+import { FitIcon, MinusIcon, PlusIcon, TargetIcon } from './lib/svgs/icons';
+import { createGraph, getElementsDefinition } from './graph';
+import type { EdgeData, EdgeStyleProps, NodeData, NodeStyleProps } from './types';
+import Container from '@awsui/components-react/container';
+import { createClassName, ClassName } from './lib/utils/element';
+
+import styles from './styles.module.scss';
+
+const GRAPH_CANVAS_PADDING = 30;
+interface ICytoNetworkProps {
+  className?: ClassName;
+  nodeData: Map<string, NodeData>;
+  edgeData?: Map<string, EdgeData> | null;
+  graphLayout?: LayoutOptions;
+  nodeStyle?: NodeStyleProps;
+  nodeHoverStyle?: NodeStyleProps;
+  nodeSelectedStyle?: NodeStyleProps;
+  edgeStyle?: EdgeStyleProps;
+  edgeHoverStyle?: EdgeStyleProps;
+  edgeSelectedStyle?: EdgeStyleProps;
+}
+
+export const CytoNetwork: React.FC<ICytoNetworkProps> = (props) => {
+  const [graph, setGraph] = useState<ReturnType<typeof createGraph>>();
+  const cytoRef = useRef<HTMLDivElement>(null);
+
+  const handleCenterClick = useCallback(() => {
+    graph?.center();
+  }, [graph]);
+
+  const handleFitClick = useCallback(() => {
+    graph?.fit(undefined, GRAPH_CANVAS_PADDING);
+  }, [graph]);
+
+  const handleZoomInClick = useCallback(() => {
+    if (graph) {
+      const currentScale = graph.getZoom();
+      graph.setZoom(currentScale + 0.1);
+    }
+  }, [graph]);
+
+  const handleZoomOutClick = useCallback(() => {
+    if (graph) {
+      const currentScale = graph.getZoom();
+      graph.setZoom(currentScale - 0.1);
+    }
+  }, [graph]);
+
+  useEffect(() => {
+    if (graph) {
+      graph.resize();
+      graph.center();
+    }
+  }, [graph]);
+
+  useEffect(() => {
+    if (cytoRef.current) {
+      const graph = createGraph(cytoRef.current, {
+        canvasPadding: GRAPH_CANVAS_PADDING,
+        fitOnLoad: true,
+        layoutOptions: props.graphLayout?.name ? props.graphLayout : undefined,
+      });
+
+      setGraph(graph);
+
+      return () => {
+        graph.dispose();
+      };
+    }
+    return undefined; //this is coz in tsconfig "noImplicitReturns" is set to  true, usually no return needed.
+  }, [props.graphLayout]);
+  useEffect(() => {
+    if (graph) {
+      if (props.edgeData) {
+        graph.setGraphData(getElementsDefinition([...props.nodeData.values()], [...props.edgeData.values()]));
+      } else {
+        graph.setGraphData(getElementsDefinition([...props.nodeData.values()]));
+      }
+      graph.resize();
+      graph.center();
+    }
+  }, [graph]);
```

Comment:
This may need props.nodeData and props.edgeData as dependencies for if the props change like in a case where multiple queries are run in the same session.

Reply:
great catch, I wouldn’t have noticed it till I started integrating queries. Fixed.